pkg/apis/maistra/v2/servicemeshcontrolplane_types.go

```diff
@@ -202,3 +202,14 @@ func (s ControlPlaneSpec) IsClusterScoped() (bool, error) {
 	}
 	return controlPlaneMode == ControlPlaneModeValueClusterScoped, nil
 }
+
+func (s ControlPlaneSpec) GetControlPlaneMode() (string, error) {
+	isClusterScoped, err := s.IsClusterScoped()
+	if err != nil {
+		return "", err
+	}
+	if isClusterScoped {
+		return ControlPlaneModeValueClusterScoped, nil
+	}
+	return ControlPlaneModeValueMultiTenant, nil
+}
```

Comment:
This needs two changes:
- This shouldn't return the two constants, but fixed values "ClusterWide" or "MultiTenant", as the constants in 2.4 will have different values.
- The `s.IsClusterScoped()` call doesn't work when cluster-wide mode is enabled in an SMCP profile. This is a general problem that doesn't just affect this PR. Let's pair on this, so I can help you figure out how to make this work with profiles.